torch/nn/modules/loss.py

```diff
@@ -262,20 +262,22 @@ class KLDivLoss(_Loss):
     and is often useful when performing direct regression over the space of
     (discretely sampled) continuous output distributions.
 
-    As with `NLLLoss`, the `input` given is expected to contain
-    *log-probabilities*, however unlike `ClassNLLLoss`, `input` is not
+    As with :class:`~torch.nn.NLLLoss`, the `input` given is expected to contain
+    *log-probabilities*. However, unlike :class:`~torch.nn.NLLLoss`, `input` is not
     restricted to a 2D Tensor, because the criterion is applied element-wise.
+    The targets are given as *probabilities* (i.e. without taking the logarithm).
 
     This criterion expects a `target` `Tensor` of the same size as the
     `input` `Tensor`.
 
-    The loss can be described as:
+    The unreduced (i.e. with :attr:`reduce` set to ``False``) loss can be described as:
 
     .. math::
-        \ell(x, y) = L = \{l_1,\dots,l_N\}^\top, \quad
-        l_n = y_n \odot \left( \log y_n - x_n \right),
+        l(x,y) = L := \{ l_1,\dots,l_N \}, \quad
+        l_n = y_n \cdot \left( \log y_n - x_n \right),
```

Comment:
per my understanding, `y_n`, `x_n`, and `l_n` are vectors (unintuitively). So, I feel the Hadamard product is correct?

Reply:
They are scalars because n is a multiindex:
> where the index \(N\) spans all dimensions of input

Reply:
Ah, good idea :) Nice :)